### ISSUE TITLE
Custom image sizes support in Oxygen editor

### ISSUE THREAD

**Issue by srikat** [soflyy/oxygen-bugs-and-features#675]
Custom image sizes (registered using `add_image_size`) do not appear in the Size dropdown for images.

![screenshot-1](https://d.pr/i/C25Tga+)

![screenshot-1](https://d.pr/i/OPg3ZZ+)

Please consider fixing/adding this.

It would also be good if custom image sizes are read and shown in all other places where image size can be selected. Ex.: https://github.com/soflyy/oxygen-bugs-and-features/issues/433.

**Comment by adrien-robert** [comment]
+1

**Comment by WesternWeb** [comment]
Please

**Comment by vicento** [comment]
+1

**Comment by PeterCarsten** [comment]
Would be helpfull

**Comment by yasien-s** [comment]
👍🏼

**Comment by jornmo** [comment]
Yes thank you!

**Comment by cliffhewson** [comment]
+1 Seems pretty fundamental!

**Comment by c001os** [comment]
+1 would be great!

**Comment by mo-mono** [comment]
+1 👌🏼

**Comment by davidlower8** [comment]
Yes please. +1.

**Comment by websmx** [comment]
Please, please!

**Comment by sunnyt7** [comment]
+1

**Comment by bpages383** [comment]
+1 it would be great

**Comment by wplit** [comment]
This is already working, but the only sizes that appear in the dropdown are specific to the image currently selected. So usually regenerating thumbnails after creating the custom image size then refreshing Oxygen will make it appear. 

![oxygen-image-size-dropdown](https://user-images.githubusercontent.com/43051571/77868093-97ba3080-7285-11ea-809b-023f3cfd15cb.jpg)

I just mentioned at the bottom of this post..

https://wpdevdesign.com/using-responsive-srcset-images-custom-image-sizes-in-oxygen/

**Comment by srikat** [comment]
Custom image sizes should also be available to pick in other places.

https://github.com/soflyy/oxygen-bugs-and-features/issues/433

**Comment by PaRoxUs** [comment]
+1

**Comment by clockwood** [comment]
+1

**Comment by igmarketing** [comment]
Yes please! Core Web Vitals will become a ranking factor in May 2021... Oxygen is really liked in the SEO community for it's speed and optimization for SEO.

![image](https://user-images.githubusercontent.com/77027887/103704375-30997d00-4f77-11eb-8969-d2edd58403c4.png)

Please don't ignore this issue guys. Thanks much!

**Comment by derSebastian** [comment]
+1

**Comment by Jehu** [comment]
What the....!!! This is still an open Bug?

**Comment by yoshi-12321** [comment]
+1

**Comment by kevingeary** [comment]
+1

**Comment by ukdesignservices** [comment]
+1

**Comment by norman-pixelkings** [comment]
+1

**Comment by nil80** [comment]
+1

**Comment by norman-pixelkings** [comment]
+1

**Comment by heavestudio** [comment]
+1000

**Comment by LauGau** [comment]
+1

**Comment by shock2d** [comment]
This is must for speed workflow

**Comment by adrien-robert** [comment]
+1

**Comment by CasualProtagonist** [comment]
++

**Comment by acicovic** [comment]
To all +1 posters: Also click on the thumbs up icon in the first post to make this go higher when sorting by thumbs up.

Also, check #433 which is related.

**Comment by YammYcoding** [comment]
+1

**Comment by gerogr** [comment]
+1

**Comment by fifle** [comment]
+1. This feature is needed a lot :)

**Comment by d-packs** [comment]
> This is already working, but the only sizes that appear in the dropdown are specific to the image currently selected. So usually regenerating thumbnails after creating the custom image size then refreshing Oxygen will make them appear in the dropdown (if that image has this size available)

This is not correct in my experience.  For some reason the image sizes are note populated, even after regenerating thumbs. This feature does not work.

**Comment by jevgen** [comment]
+1. Needed a lot.

**Comment by d-packs** [comment]
> This is already working, but the only sizes that appear in the dropdown are specific to the image currently selected. 
Even if it would have worked like this, this is not the correct behavior. The drop-down should not be limited to the currently selected image sizes. Images and their sizes are in continuous flux during development. If an image size has been registered with `add_image_size` it should appear in the dropdown.

**Comment by jevgen** [comment]
Thank you @d-packs , it really does. I added custom image size declaration to custom functionality plugin and it worked. The only hinder was to find out an appropriate way to regenerate thumbnails. And built in WP-CLI support for that helped a lot. WP-CLI can selectively regenerate thumbnails for array of IDs.

**Comment by vadimbk** [comment]
Add this to CodeSnippets and activate
```
add_theme_support( 'post-thumbnails' );
add_image_size( 'image-480', 480, 9999 );
add_image_size( 'image-640', 640, 9999 );
add_image_size( 'image-720', 720, 9999 );
add_image_size( 'image-960', 960, 9999 );
add_image_size( 'image-1168', 1168, 9999 );
add_image_size( 'image-1440', 1440, 9999 );
add_image_size( 'image-1920', 1920, 9999 );

add_filter( 'image_size_names_choose', 'custom_image_sizes' );

function custom_image_sizes( $size_names ) {
$new_sizes = array(
'image-480' => __( 'Size480', 'text-domain' ),
'image-640' => __( 'Size640', 'text-domain' ),
'image-720' => __( 'Size720', 'text-domain' ),
'image-960' => __( 'Size960', 'text-domain' ),
'image-1160' => __( 'Size1160', 'text-domain' ),
'image-1440' => __( 'Size1440', 'text-domain' ),
'image-1920' => __( 'Size1920', 'text-domain' )
);
return array_merge( $size_names, $new_sizes );
}
```
Unfortunately it doesn't work for repeaters

**Comment by vadimbk** [comment]
To use proper file size in dynamic data insert a function in code snippets
/**
* Returns the file name of right size of image
* @param  string $size registered name of image size
* @return string        proper filename of image .
*/
function right_image_filename( $size ) {

return wp_get_attachment_image_src(get_post_thumbnail_id($post->ID), $size)[0];
}

in data dialog select php function

insert 2 parameters:
Function name **right_image_filename**
Name of registered size **image-489**

and voila 
Works like a charm

**Comment by nathanbitcheno** [comment]
> To use proper file size in dynamic data in repeater insert a function in code snippets
> /**
> 
> * Returns the file name of right size of image
> * @param  string $size registered name of image size
> * @return string        proper filename of image .
>   */
>   function right_image_filename( $size ) {
> 
> return wp_get_attachment_image_src(get_post_thumbnail_id($post->ID), $size)[0];
> }
> 
> in data dialog select Advanced (php function return value)
> 
> insert 2 parameters:
> Function name **right_image_filename**
> Name of registered size **image-489**
> 
> and voila
> Works like a charm

This works wonderfully. Excellent work.

**Comment by soterogouveia** [comment]
vadimbk's answer worked perfecty!

Anyone found a workaround for using the same custom image sizes for the background images of sections and DIVs? 

This would be equally useful!

Thank you

**Comment by elmichalos** [comment]
Oh yes much needed

**Comment by DrZivil** [comment]
+1

**Comment by Hefestos21** [comment]
> To use proper file size in dynamic data in repeater insert a function in code snippets /**
> 
> * Returns the file name of right size of image
> * @param  string $size registered name of image size
> * @return string        proper filename of image .
>   */
>   function right_image_filename( $size ) {
> 
> return wp_get_attachment_image_src(get_post_thumbnail_id($post->ID), $size)[0]; }
> 
> in data dialog select Advanced (php function return value)
> 
> insert 2 parameters: Function name **right_image_filename** Name of registered size **image-489**
> 
> and voila Works like a charm

Hi! Sorry, but I must be very dumb! I can't follow those instructions 😢
I placed the snippet (literally.. or should I change anything)
And then in the PHP function return value, I can see the Function name, but where do I place the "image-489" part?

please help
Thanks in advance!

**Comment by MimiDev** [comment]
I just saw this thread here and I've been using this bit of code (that I add in Code Snippets) which adds the custom image sizes to the Oxygen Builder options. I can't actually remember where I found it, but maybe it's of help to others?

`add_image_size( 'custom-image-1',500,325, true );
add_image_size( 'custom-image-2',1920,400, true );

/* use this to insert with PHP
	<?php the_post_thumbnail( 'your-specified-image-size' ); ?>
	
	*** For use with the Oxy 'PHP function return value' data field ***
	<?php get_the_post_thumbnail_url( ,'your-specified-image-size' ); ?>
	*/

// Add our custom sizes to list of default image sizes
add_filter( 'image_size_names_choose', 'lit_merge_images_sizes' );
function lit_merge_images_sizes( $sizes ) {
    return array_merge( $sizes, array(
        'custom-image-1' => __( 'Custom Image One' ),
		'custom-image-2' => __( 'Custom Image Two' ),
    ) );
}`

**Comment by gik000** [comment]
I tried everything, but nothing worked.

I done it through an active custom plugin, but it is not "considered" in oxygen builder back-end interface.
Even a full thumbnail regeneration, hasn't fixed the problem.
